imblearn/under_sampling/_prototype_selection/_instance_hardness_threshold.py

```diff
@@ -126,6 +126,9 @@ def _validate_estimator(self):
                 isinstance(self.estimator, ClassifierMixin) and
                 hasattr(self.estimator, 'predict_proba')):
             self.estimator_ = clone(self.estimator)
+            self.estimator_.set_params(random_state=self.random_state)
+            if 'n_jobs' in self.estimator_.get_params().keys():
```

Comment:
Since you are trying to set one parameter I think that it is more pythonic and [easier to ask for forgiveness than permission](https://docs.python.org/3/glossary.html#term-eafp).